File: VotingApp_Client_WEB/Models/HttpPostRequest.js
// Define the HttpPostRequest class
export class HttpPostRequest {
    constructor(session, user) {
      this._session = session;
      this._user = user;
    }
  
    // Getter and setter for VotingSession property
    get VotingSession() {
      return this._session;
    }
  
    set VotingSession(session) {
      this._session = session;
    }
  
    // Getter and setter for User property
    get User() {
      return this._user;
    }
  
    set User(user) {
      this._user = user;
    }

    // Method to convert the object to JSON with custom property names
    toJSON() {
        return {
            votingSession: this._session,
            voter: this._user
        };
    }
  }
  

File: VotingApp_Client_WPF_non_functional/HttpPostRequest.cs
using System.Text.Json.Serialization;

namespace VotingApp_Client_WPF
{
    // wrapper class for http request
    internal class HttpPostRequest
    {
        private VotingSessionEgress _session;
        private VoterEgress _user;

        public HttpPostRequest(VotingSessionEgress session, VoterEgress user)
        {
            _session = session;
            _user = user;
        }

        [JsonPropertyName("votingSession")]
        public VotingSessionEgress VotingSession { get { return _session; }  set { _session = value; } }
        [JsonPropertyName("voter")]
        public VoterEgress User { get { return _user; } set { _user = value; } }

    }
}
